iceoryx_hoofs/include/iceoryx_hoofs/error_handling/error_handling.hpp

```diff
@@ -209,45 +202,18 @@ enum class ErrorLevel : uint32_t
     MODERATE
 };
 
-using HandlerFunction = std::function<void(const Error error, const std::function<void()>, const ErrorLevel)>;
-
-/// @brief This handler is needed for unit testing, special debugging cases and
-///         other corner cases where we'd like to explicitly suppress the
-///         error handling.
-class ErrorHandler
-{
-    friend void
-    errorHandler(const Error error, const std::function<void()>& errorCallBack, const ErrorLevel level) noexcept;
-
-  public:
-    static cxx::GenericRAII setTemporaryErrorHandler(const HandlerFunction& newHandler) noexcept;
-
-    static const char* toString(const Error error) noexcept;
-
-  protected:
-    static void reactOnErrorLevel(const ErrorLevel level, const char* errorText) noexcept;
-
-  private:
-    static void defaultHandler(const Error error,
-                               const std::function<void()>& errorCallBack,
-                               const ErrorLevel level = ErrorLevel::FATAL) noexcept;
-
-    static const char* ERROR_NAMES[];
-    static iox::HandlerFunction handler;
-    /// Needed, if you want to exchange the handler. Remember the old one and call it if it is not your error. The error
-    /// mock needs to be the last one exchanging the handler in tests.
-    static std::mutex handler_mutex;
-};
-
 /// @brief Howto use the error handler correctly
-///     1.) If the error you would like to handle is not listed in ICEORYX_ERRORS(error)\...
-///             macro just add them like:
+///     1.) Use the macro ICEORYX_ERRORS(error)\ to create the enum for your component and
+///             add new errors like:
 ///             error(MODULE_NAME__MY_FUNKY_ERROR)
 ///         Attention: Create an error after the following convention:
 ///             MODULE_NAME__A_CLEAR_BUT_SHORT_ERROR_DESCRIPTION
 ///         And a long name is alright!
 ///
-///     2.) Call errorHandler(Error::kMODULE_NAME__MY_FUNKY_ERROR);
+///     2.) Specialize the following methods for your NewEnumErrorType:
+///         - const char* toString(const NewEnumErrorType error)
+///
+///     3.) Call errorHandler(Error::kMODULE_NAME__MY_FUNKY_ERROR);
```

Comment:
There wasn't a decision on that topic yet. If we will change it, I would for sure do it in a follow-up PR.

Reply:
@elBoberido @mossmaurice I think we introduced the `k` prefix so that it conforms to the autosar guidelines and then we found out that this is actually pretty nice since otherwise we would run into some trouble with our macro.

So whoever does this - be prepared for fun :D

Reply:
@elfenpiff that's no problem. One has just to replace `#define CREATE_ICEORYX_ERROR_ENUM(name) k##name,` with `#define CREATE_ICEORYX_ERROR_ENUM(name) name,`